Makefile.am

```diff
@@ -1,13 +1,15 @@
 ACLOCAL_AMFLAGS = -I libltdl/m4
 
-SUBDIRS = src/libmongoc
+if BUILD_WITH_OWN_LIBMONGOC
+LIBMONGOC_SUBDIR=src/libmongoc
+endif
+
+SUBDIRS = $(LIBMONGOC_SUBDIR) proto src bindings .
```

Comment:
I gave this a try and ran into issues. This is because the sub directories are compiled in the [listed order](https://www.gnu.org/software/automake/manual/html_node/Subdirectories.html). Since the Makefile under `src` has [dependencies on MongoDB](https://github.com/Stackdriver/collectd/blob/stackdriver-agent-5.5.2/src/Makefile.am#L716), the build is failing due to missing shared objects.

Any thoughts on how to proceed? I noticed we preserve the same ordering in [our current Makefile](https://github.com/Stackdriver/collectd/blob/stackdriver-agent-5.5.2/Makefile.am#L3). I can try to find another approach if this is undesirable though.

Reply:
I would do it this way, then:
```
SUBDIRS =

if BUILD_WITH_OWN_LIBMONGOC
SUBDIRS += src/libmongoc
endif

if BUILD_INCLUDED_LTDL
SUBDIRS += libltdl
endif

SUBDIRS += proto src bindings .
```
WDYT?